test/src/specs/relay/compact_block.rs

```diff
@@ -152,13 +139,66 @@ impl CompactBlockBasic {
 
         clear_messages(net);
     }
+
+    pub fn test_lose_get_block_transactions(
+        &self,
+        net: &Net,
+        node0: &Node,
+        node1: &Node,
+        peer_id0: PeerIndex,
+    ) {
+        node0.generate_block();
+        let new_tx = node0.new_transaction(node0.get_tip_block().transactions()[0].hash().clone());
+        node0.submit_block(
+            &node0
+                .new_block_builder(None, None, None)
+                .proposal(new_tx.proposal_short_id())
+                .build(),
+        );
+        // Proposal a tx, and grow up into proposal window
+        node0.generate_blocks(6);
+
+        // Make node0 and node1 reach the same height
+        node1.generate_block();
+        node0.connect(node1);
+        node0.waiting_for_sync(node1, node0.get_tip_block().header().number(), 20);
+
+        // Net consume and ignore the recent blocks
+        clear_messages(net);
+
+        // Construct a new block contains one transaction
+        let block = node0
+            .new_block_builder(None, None, None)
+            .transaction(new_tx)
+            .build();
+
+        // Net send the compact block to node0, but dose not send the corresponding missing
+        // block transactions. It will make node0 unable to reconstruct the complete block
+        net.send(
+            NetworkProtocol::RELAY.into(),
+            peer_id0,
+            build_compact_block(&block),
+        );
+        let (_, _, data) = net
+            .receive_timeout(Duration::from_secs(10))
+            .expect("receive GetBlockTransactions");
+        let message = get_root::<RelayMessage>(&data).unwrap();
+        assert_eq!(
+            message.payload_type(),
+            RelayPayload::GetBlockTransactions,
+            "Node0 should send GetBlockTransactions message for missing transactions",
+        );
+
+        // Submit the new block to node1. We expect node1 will relay the new block to node0.
+        node1.submit_block(&block);
+        node0.waiting_for_sync(node1, node0.get_tip_block().header().number(), 20);
```

Comment:
node1's tip number is higher than node0, should use node1 tip number as the target, this test will fail randomly otherwise.

```suggestion
        node0.waiting_for_sync(node1, node1.get_tip_block().header().number(), 20);
```